### ISSUE TITLE
Magento 2.0.4 running painfully slow on Wamp, unable to load pages or save settings: Error: client denied by server configuration app/etc/config.php

### ISSUE THREAD

**Issue by shashankitsoft** [magento/magento2#4363]
## Steps to reproduce
1. Installed Magento 2.0.4 on WAMP 3.0 on windows 10 Pro 32 bit, 4 GB Ram, PHP 5.6.15, My SQL 5.7.9, Apache 2.4.17. My localhost base url is: http://127.0.0.1:8080/mymagento/
2. The new magento installation worked fine for few days, But now suddenly it became **too slow... Not able to load pages in frontend and unable to save settings in admin  too. Browser just keep waiting..**
3. Checked apache error log... These errors are coming every time I am refreshing my localhost url..

```
[access_compat:error] [pid 14060:tid 964] [client 127.0.0.1:57098] AH01797: client denied by server configuration: E:/wamp/www/mymagento/app/etc/config.php
[access_compat:error] [pid 2336:tid 1088] [client 127.0.0.1:57101] AH01797: client denied by server configuration: E:/wamp/www/mymagento/app/etc/config.php
```
1. Already cleared var and pub/static folders, restarted wamp several times. Also, ran commands 

```
php bin/magento indexer:reindex
php bin/magento setup:upgrade
php bin/magento maintenance:disable
```
1. In fact this issue came after I ran command `php bin/magento setup:upgrade` last time, although I have used this command earlier too with no issues. So, not confirm this is the cause.

_Note:_ Much time earlier when I ran `php bin/magento setup:static-content:deploy` command on a separate magento 2 installation this (painfully slow) issue came and I had to reinstall this magento when no other solution found. For the same fear, I am also not enabling merge css/js setting in magento admin this time (which I did in old installation, suspecting it a cause #3651 )

_Note 2:_ I don't think this is php memory_limit or execution_time issue. bcoz I have already increased it sufficient, and was working fine on these settings until this slowness issue appeared. Further, I have already set few  (not all) settings as per guide http://www.devside.net/wamp-server/wamp-is-running-very-slow to optimize the WAMP & mysql performance.
## Actual result

Painfully slow, without a identifiable reason. Unable to load pages in frontend and unable to save settings in admin  too. Browser just keep waiting.
(Although I have found, occasionally it runs well sometime. Weird!)

Any solution?

Thanks.

**Comment by JDavidVR** [comment]
I was having same issue, the problem is that magento 2 is designed for php7 so my best recommendation is that you should upgrade your web server I recommend you to install it on linux. That solves the problem.

guide to install webserver on linux, with largest versions of PHP mysql apache and phpmyadmin
http://jdavidvr.github.io/webserver2016/

**Comment by shashankitsoft** [comment]
I am not agree with you. Magento 2 works on php 5.6 and on 7 too. It requires minimum mysql 5.6. Also, Most of the shared hosting these days still not upgraded to php 7 till now.
Installing it on linux is also **not a solution**.

There is some problem with magento 2, and that need to fixed for both windows and linux platforms. Development can be done on any platform.

Thanks, but no thanks.

**Comment by Yogesh7989** [comment]
Hello Guys,

Set Developer mode on get details from here http://devdocs.magento.com/guides/v2.0/config-guide/cli/config-cli-subcommands-mode.html

Remove comment from magento 2  htaccess file from line number 5 (#   SetEnv MAGE_MODE developer).
make memory_limit = 2024mb for local
Enable Cache.

Its working fine in my local system.

Please let me know if still not working

Thanks

**Comment by adragus-inviqa** [comment]
> the problem is that magento 2 is designed for php7  

No, it's not. 

PHP 7 wasn't even released when M2 was in beta. [Besides...](https://github.com/magento/magento2/blob/ee6159109fa47fef1f13a71d2cec69fc8703291b/composer.json#L11)

**Comment by JDavidVR** [comment]
I'm sorry, I think that I used and incorrect word, I mean magento is optimized for php7, I was using php5.6 and to view a product page or category it took like 30-40 seconds(checkout loaded in 5mins) I upgraded to php7 and same pages took less than 10 seconds to load (maybe is due that php7 faster than 5.x) but that is my experience.

I tried many solutions wamp xampp lamp ampps on windows mac and linux, so what worked for me is:
virtualize linux, and install each package manually.

I agree with you that the development should be the same in most of platforms, but according to my experience developing in linux is better, but that is relative I have friends that prefers windows, mac, others prefer to use solutions like vagrant.

**Comment by shashankitsoft** [comment]
Mr. Yogesh,

Comment is already removed from that htaccess line.
Memory limit already enough.
Cache etc. already cleared.

It also working fine on my local system, this issue arise many times and **suddenly become everything stuck, unable to load pages**... and apache gives **_error: client denied by server configuration app/etc/config.php_**  

Windows already have full file/folder permissions. The issue is something related to permission with localhost url, Port (127.0.0.1:8080), Dns, config, or may be some internal way how magento trying to get data. That unknown reason we have to find. 
(Btw, also checked with disabling window firewall, antivirus, cleared browser cookies/history/cache)

[FYI: I have not setup any virtual host (I don't need it), so running simply on base url: http://127.0.0.1:8080/mymagento/]

Sometimes one day it works with the same settings, sometimes the other day it stucks for indefinite time. That's the weird.

Thanks.

**Comment by mazhalai** [comment]
@shashankitsoft Windows is not officially supported, only Linux distributions are supported for production mode.  See http://devdocs.magento.com/guides/v2.0/install-gde/system-requirements.html

**Comment by shashankitsoft** [comment]
@mazhalai When magento installs it checks for prerequisites, when all readiness checks okay, only then it installs further. By that way it properly installs, run and never shows warning for windows platform not supported. If not fully official, it is not unofficial too. 
Further, I am currently developing on local windows, so its _not in production mode_, its in developer mode. Will host later on linux/CentOS platofrm somehwere domain in production mode ofcourse.

http://devdocs.magento.com/guides/v2.0/install-gde/install/web/install-web.html

I worked on wamp/windows, so the issue what came I reported. I haven't checked linux, so can't say. May be sometimes linux may have the same issue, don't know.

My focus is on the issue, not the platform: why its coming, what can be done to resolve immediately when it comes.. 

Today also, it stopped working for 2-3 hours and got another apache error:`zend_mm_heap corrupted` along with `client denied error` and later worked itself (after several restarts of wamp & windows, cleaning magento etc.). Ghost problem. Wierd.

Thanks.

**Comment by mazhalai** [comment]
@shashankitsoft the reason this check is not in readiness check is that windows is unofficially supported only for developers. This in no way means we support running a store in production on Windows.  

Re `zend_mm_heap` error, did you try this? http://stackoverflow.com/questions/2247977/what-does-zend-mm-heap-corrupted-mean

**Comment by shashankitsoft** [comment]
@mazhalai I have updated my above reply.. see. 

> I am currently developing on local windows, so its _not in production mode_, its in **developer mode**. Will host later on linux/CentOS platofrm somehwere domain in production mode ofcourse.

Yes I have already checked that link for `zend_mm_heap` problem, I have disabled `output_buffering` once, restarted wamp, run fine, then enabled again , and the error gone now.

Later after struggling, (as I mentioned earlier) its currently in 'working fine' status. Until I will get the same problem again `client denied error` may be tomorrow again.. can't say.

Thanks

**Comment by piotrekkaminski** [comment]
API test MAGETWO-1234

**Comment by mazhalai** [comment]
Closing as not a bug.

**Comment by shashankitsoft** [comment]
@mazhalai  Don't know why  you closed it, when I clarified to you its in 'developer' mode not 'production' mode. (the issue still exists)

**Comment by piotrekkaminski** [comment]
@shashankitsoft the client denied error is not a bug. It is a security check that i think runs every time you open admin and if it can access the file, it raises a warning (that config is accessible outside).

**Comment by noc2spam** [comment]
Why is this not a bug? Sometimes in the developer mode, it takes 5 minutes to load a single page. That's crazy! It's a clean install with nothing extra.

Using WAMP ( php5.6.4) in windows. Allocated 1G of RAM, but I do not have a SSD drive handy right now.

**Comment by maconeto** [comment]
confirmed. same environment, developer mode, magento 1 takes 2s to load a page while magento 2 takes 12s

**Comment by al1357** [comment]
Same story here. Magento 2.1.10 + Ubuntu + php7.0 in virtual box and page load can be measured in minutes or pages do not load at all. Vagrant syncing is disabled, virtual box is run via vagrant or standalone - no difference.

I have exactly the same magento installation on a hosting company's server and it works fine. I've installed all required extensions for php. I've also installed a clean version of Magento - it finished quickly and without errors, all reqirements were met. Only after I loaded homepage/admin page, it stopped working.

Regarding xampp it doesn't work on it either(xampp freeze + exception) in my case so I'm not even trying.
